kubernetes/linux/main.sh

```diff
@@ -190,8 +190,8 @@ if [ -e "/etc/omsagent-secret/WSID" ]; then
             # extract the port
             port="$(echo $hostport | sed -e 's,^.*:,:,g' -e 's,.*:\([0-9]*\).*,\1,g' -e 's,[^0-9],,g')"
 
-            if [ -z "$user" -o -z "$pwd" -o -z "$host" -o -z "$port" ]; then
-               echo "-e error proxy endpoint should be in this format http(s)://<user>:<pwd>@<hostOrIP>:<port>"
+            if [ -z "$host" -o -z "$port" ]; then
+               echo "-e error proxy endpoint should be in this format http(s)://<hostOrIP>:<port> or http(s)://<user>:<pwd>@<hostOrIP>:<port>"
```

Comment:
where do we check if the ca cert given is indeed a valid cert object ?

Reply:
> where do we check if the ca cert given is indeed a valid cert object ?

> the cert validation should be taken care of the AKS RP since passed via aks.

Reply:
for non-aks ?

Reply:
> for non-aks ?

it will be taken care of the Arc k8s platform since these being based by the RPs.

Reply:
i hope they are getting validated. Because i see we just mount this into /etc/ssl which has global (container wide) impact. They could be some scripts that can harm us as well.

Reply:
> i hope they are getting validated. Because i see we just mount this into /etc/ssl which has global (container wide) impact. They could be some scripts that can harm us as well.

> yes. these be are being validated at the RP since  these are being used for other cluster components also.

Reply:
i checked , and the only validation happening is base64 encoding. So you can wrap anything as base64 string and get that landed in /etc/ssl/proxycert.crt